### PR TITLE
Improve handling of month + year datetime cases

### PR DESCRIPTION
**Pull request by guom08**
Issue #315. 

"January , 2016"
"2016 January"

Also, the values below work without spaces. They should also work with spaces across languages:
"1995 - 01"
"1995 / 01"
"01 - 1995"
"01 / 1995"

Supported in English, German and French. The last four cases are not fully supported in Chinese, Portuguese and Spanish in this PR.

### CODE REVIEW

File: .NET/Samples/SimpleConsole/Program.cs.rej [hunk 1]
```diff
@@ -0,0 +1,9 @@
+diff a/.NET/Samples/SimpleConsole/Program.cs b/.NET/Samples/SimpleConsole/Program.cs	(rejected hunks)
+@@ -88,6 +88,7 @@
+                 // E.g "Set the temperature to 30 degrees celsius" will return "30 C"
```
Comment (tellarin) [inline .NET/Samples/SimpleConsole/Program.cs.rej]: What is this?

File: .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs [hunk 2]
```diff
@@ -45,9 +45,9 @@ public static class DateTimeDefinitions
...
 		public static readonly string BetweenRegex = $@"\b(between\s+)({DayRegex})\s*{RangeConnectorRegex}\s*({DayRegex})\s+{MonthSuffixRegex}((\s+|\s*,\s*){YearRegex})?\b";
-		public static readonly string MonthWithYear = $@"\b(({EngMonthRegex},?(\s+of)?\s+({YearRegex}|(?<order>next|last|this)\s+year))|(({YearRegex}|(?<order>next|last|this)\s+year)\s*,?\s+{EngMonthRegex}))\b";
+		public static readonly string MonthWithYear = $@"\b(({EngMonthRegex}(\s*),?(\s+of)?(\s*)({YearRegex}|(?<order>next|last|this)\s+year))|(({YearRegex}|(?<order>next|last|this)\s+year)(\s*),?(\s*){EngMonthRegex}))\b";
 		public static readonly string OneWordPeriodRegex = $@"\b((((the\s+)?month of\s+)?({RelativeRegex}\s+)?(?<month>April|Apr|August|Aug|December|Dec|February|Feb|January|Jan|July|Jul|June|Jun|March|Mar|May|November|Nov|October|Oct|September|Sep|Sept))|(month|year) to date|({RelativeRegex}\s+)?(weekend|week|month|year)(\s+{AfterNextSuffixRegex})?)\b";
-		public static readonly string MonthNumWithYear = $@"({YearNumRegex}[/\-\.]{MonthNumRegex})|({MonthNumRegex}[/\-]{YearNumRegex})";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: Did you run some perf tests to see if there was any impact?

Comment (guom08) [reply .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: I've done the perf test and the upper bound of time increase is 0.3%.

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs]: After offline chat, it seems the impact is much small than this. Good. :)

**Pull request merged**